### ISSUE TITLE
Search results won't close when clicked outside the control

### ISSUE THREAD

**Issue by bernard4uece** [ghoshnirmalya/react-search-box#99]
Please see the screenshot. I searched the text but didn't select any value and I clicked on white space in the page. And the search results box remains open.
![image](https://user-images.githubusercontent.com/41011210/83334312-9ac8a180-a273-11ea-8e85-2a216d45689b.png)

**Comment by pascuflow** [comment]
@ghoshnirmalya I agree, I'm making a button next to the box that when clicked will redirect to a new page with the keywords on the input box, can you just give us a Clear props similar to clearOnSelect?

**Comment by pascuflow** [comment]
For anyone in a hurry or wants to make an pr, add this funtion to src/index.tsx:
```
function useOutsideAlerter(ref, setDropdownVisibility, setValue) {
  useEffect(() => {
    /**
     * Alert if clicked on outside of element
     */
    function handleClickOutside(event) {
      if (ref.current && !ref.current.contains(event.target)) {
        setDropdownVisibility(false)
        setValue('')
      }
    }

    // Bind the event listener
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      // Unbind the event listener on clean up
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, [ref]);
}
const ReactSearchBox: FC<IProps> = ({
...
...
...
  const [matchedRecords, setMatchedRecords] = useState<any>([]);
  const [value, setValue] = useState<string>("");
  const [showDropdown, setDropdownVisibility] = useState<boolean>(false);
  const wrapperRef = useRef(null);
  useOutsideAlerter(wrapperRef, setDropdownVisibility, setValue);
  
 .....
  return (
    <div ref={wrapperRef}>
      <StyledContainer>
        <GlobalStyle />
        {inputNode()}
        {dropdownNode()}
      </StyledContainer>
    </div>
  );
```

**Comment by ghoshnirmalya** [comment]
@pascuflow Thank you for the suggestion.

This issue has been fixed in [2.2.1](https://github.com/ghoshnirmalya/react-search-box/releases/tag/v2.2.1). The [demo](https://17ml6.csb.app/) has been updated as well to show this bug fix.
